public/js/specberus.js

```diff
@@ -192,35 +199,62 @@ jQuery.extend({
         attachCustomScroll();
         // endValidation();
     });
+    socket.on("finishedExtraction", function (data) {
+      $(".manual").toggle(true);
+      var processDocument = (data.processdocument && data.processDocument.indexOf('Process-20051014') > -1) ? "2005" : "2015";
+      $profile.val(data.profile);
+      $processDocument.find("label#2005").toggleClass("active", (processDocument === "2005"));
+      $processDocument.find("label#2015").toggleClass("active", (processDocument === "2015"));
+      $noRecTrack.prop('checked', !data.rectrack);
+      $informativeOnly.prop('checked', data.informative);
+      $validation.val('simple-validation');
+      var options = {
+                        "url"             : data.url
+                      , "profile"         : data.profile
+                      , "validation"      : 'simple-validation'
+                      , "noRecTrack"      : !data.rectrack || false
+                      , "informativeOnly" : data.informative || false
+                      , "echidnaReady"    : false
+                      , "patentPolicy"    : "pp2004"
+                      , "processDocument" : processDocument
+                    };
+      validate(options);
+      var newurl = document.URL.split('?')[0] + "?" + $.param(options)
+      history.pushState(options, url + " - " + profile, newurl);
+    });
 
     // handle the form
     $("#options").submit(function () {
         clearError();
-        var url = $url.val()
-        ,   profile = $profile.val()
-        ,   validation = $validation.val()
-        ,   noRecTrack = $noRecTrack.is(":checked") || false
-        ,   informativeOnly = $informativeOnly.is(":checked") || false
-        ,   echidnaReady = $echidnaReady.is(":checked") || false
-        ,   patentPolicy = $patentPolicy.find('label.active').attr('id')
-        ,   processDocument = $processDocument.find('label.active').attr('id')
-        ;
-        if (!url) showError("Missing URL parameter.");
-        if (!profile) showError("Missing profile parameter.");
-        if (echidnaReady) profile += '-Echidna';
```

Comment:
If in the UI you tick the checkbox `#echidnaReady`, the `Auto-detect` profile gets disabled, and you can't select it again.
I think it's because in this line `-Echidna` is appended to the profile that is selected, and the only profile that exists both with and without `-Echidna` is `WD`.
Perhaps we should make an exception if the selected profile is `auto` (or have `auto-Echidna` too).